deon/cli.py

```diff
@@ -28,41 +21,20 @@
                                                                       Default is False , which will append \
                                                                       to existing file.')
 def main(checklist, format, output, clipboard, overwrite):
-    # load checklist
-    cl_path = Path(checklist) if checklist else DEFAULT_CHECKLIST
-    cl = Checklist.read(cl_path)
-
-    output = Path(output) if output else None
-
-    # output extension is given priority if differing format is included
-    if output:
-        # get format by file extension
-        ext = output.suffix.lower()
-        if ext in EXTENSIONS.keys():
-            output_format = EXTENSIONS[ext]
-        else:
-            with click.get_current_context() as ctx:
-                msg = 'Output requires a file name with a supported extension.\n'
+    try:
+        result = create(checklist, format, output, clipboard, overwrite)
+    except ExtensionException:
+        with click.get_current_context() as ctx:
+                msg = 'Output requires a file name with a supported extension.\n\n'
                 raise click.ClickException(msg + ctx.get_help())
-    elif format:
-        if format in FORMATS:
-            output_format = format
-        else:
-            with click.get_current_context() as ctx:
-                msg = "File format is not supported.\n"
+    except FormatException:
+        with click.get_current_context() as ctx:
+                msg = "File format is not supported.\n\n"
                 raise click.ClickException(msg + ctx.get_help())
     else:
-        output_format = 'markdown'
-
-    template = FORMATS[output_format](cl)
-
-    # write output or print to stdout
-    if output:
-        template.write(output, overwrite=overwrite)
-    elif clipboard:
-        xerox.copy(str(template.render()))
-    else:
-        click.echo(template.render())
+        # write output or print to stdout
+        if result:
+            click.echo(result)
```

Comment:
Can we add a `click.echo` call for clipboard and writing to a file that tell the user what happened and that it succeeded?

Reply:
In this I didn't understand exactly what you mean. Perhaps you could rephrase it?

Reply:
Something like:

```
elif clipboard:
    click.echo("Checklist successfully copied to clipboard.")
else:
    click.echo(f"Checklist successfully written to file {output}.")
```